Add link checker in CI (#266)
medic/cht-docs: README.md

1. Get local copies of the project submodules so you can build and run your site locally:
   - `git submodule update --init --recursive`
1. Build your site:
   - `hugo server`
   - If there are any errors, check that you have the latest version of Hugo, and are using the extended version.
1. Preview your site in your browser at: http://localhost:1313/
1. As you make changes to the site, your browser will automatically reload your local dev site so you can easily and quickly see your changes. 

> ## Optional Link Checking

Reviewer: Want to make sure that it's clear that _local_ link checking is optional... Is there a better way to bring in the optional sub header just before the install instructions?

```suggestion
## Link Checking [Optional]
```

Author: I feel like given this document is entirely about doing dev, that's it's obvious you're doing link checking against local host. We don't say that you install `hugo` to do local dev, nor call out that you're building docs to do local dev, but both cite `localhost:1313`

That said, I don't feel strongly about this ;)  If you think we need to change some language here to call this out, I'm game.

> We validate that all links on the docs site work (do not 404) using a tool called [Muffet](https://github.com/raviqqe/muffet) along with  [Actions](https://github.com/features/actions).  If you're creating a lot of new links, or editing a lot of existing links, you may optionally run Muffet locally before pushing your commits.  This will save a lot of time of pushing a build, waiting for the Action to run, finding you have a broken link, fixing it, pushing a new change etc.

Reviewer: Not sure if the double spaces were typos, a typing preference, or both :)

We generally have single spaces after periods so have made that correction in this suggestion. Also, the last sentence was confusing to read at first... tried to make it clearer, but could use yet another pass:

```suggestion
We validate that all links on the docs site work (do not 404) using a tool called [Muffet](https://github.com/raviqqe/muffet) along with [Actions](https://github.com/features/actions). If you're creating a lot of new links, or editing a lot of existing links, you may optionally run Muffet locally before pushing your commits. Running Muffet locally can save time by exposing broken links before pushing a build since you can avoid waiting for the Action to run, finding you have a broken link, fixing it, and pushing a new change.
```

Author: I'd say this old dog is having trouble learning new tricks ;) My typing class 30 years ago (on actual typewriters!!) has me doing double spaces after periods.  The other spaces...are just mistakes?

Either way, thanks for the space fixes.  And thanks for the re-wording - agreed this is clearer what's going on.

  - Install [Go](https://golang.org/doc/install) as a prerequisite 
  - Install Muffet: `go get -u github.com/raviqqe/muffet`
  - Ensure you've run `hugo server` so your local docs instance is reachable at http://localhost:1313/
  - Test the links with the [`muffet.sh`](https://github.com/medic/cht-docs/blob/master/.github/scripts/muffet.sh) script.  If you're in the root of this repo, that'd be: `./.github/scripts/muffet.sh` 
  
It should take about 60 seconds depending on your Internet connection. If Muffet returns no output, you have no broken links, congrats! 

_Note_: The `muffet.sh` script here is the identical script we run on GitHub. If you simply run `muffet http://localhost:1313` you will hit GitHub's rate limiting and get lots of [429's](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429). Our script intentionally reduces concurrency and excludes some repetitive GitHub URLs.

## Continuous Deployment

> All changes to `master` branch run a [GitHub action](.github/workflows/ci.yml) to first check for any broken links (per above) and then deploy to the documentation site: [docs.communityhealthtoolkit.org](https://docs.communityhealthtoolkit.org)

Reviewer: Would be nice to link "per above" to the proper section and make it easy to the reader to find it.

Author: yup yup - agreed! see 05a8355 .

 However, given you're already scrolled down to the bottom of the page, the net effect is that the page doesn't actually move at all.  It'll be nice for copying and pasting deep link though!
